### PR TITLE
[FIX] figure: Avoid scroll on a focused figure

### PR DESCRIPTION
**Pull request by rrahir**
## [FIX] figure: Avoid scroll on a focused figure 

Focusing an element will (by default) scroll the element into view (https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/focus) unfortunately, due to the current HTML architecture, the scroll will pull the whole grid overlay when trying to display a figure that overflow outside the currently visible viewport.

NOTE: The solution proposed is not supported on mobile browsers but a quick test on firefox Android showed that we do not have the same overflow issue.

To reproduce:
- Create a chart on a sheet
- copy chart (Ctrl+C)
- select a cell in the last visible row of the sheet
- paste the chart (Ctrl+V) -> the layout is broken

## [FIX] figure: apply dragging style on actual drag, not on mousedown

making the chart transparent on a simple click makes no sense

[FIX] GridOverlay: Only react to self targetted double clicks

a dbl click event targeting a child of gridOverlay will be caught by the latter and start the edition ....

Task 3090033

## Description:

description of this task, what is implemented and why it is implemented that way.

Odoo task ID : [3090033](https://www.odoo.com/web#id=3090033&action=333&active_id=2328&model=project.task&view_type=form&cids=1&menu_id=4720)

## review checklist

- [ ] feature is organized in plugin, or UI components
- [ ] support of duplicate sheet (deep copy)
- [ ] in model/core: ranges are Range object, and can be adapted (adaptRanges)
- [ ] in model/UI: ranges are strings (to show the user)
- [ ] undo-able commands (uses this.history.update)
- [ ] multiuser-able commands (has inverse commands and transformations where needed)
- [ ] new/updated/removed commands are documented
- [ ] exportable in excel
- [ ] translations (\_lt("qmsdf %s", abc))
- [ ] unit tested
- [ ] clean commented code
- [ ] track breaking changes
- [ ] doc is rebuild (npm run doc)
- [ ] status is correct in Odoo

### CODE REVIEW

File: tests/components/grid.test.ts [hunk 1]
```diff
@@ -1470,4 +1471,24 @@ describe("Copy paste keyboard shortcut", () => {
     expect(model.getters.getChartIds(sheetId)).toHaveLength(1);
     expect(model.getters.getChartIds(sheetId)[0]).not.toEqual("chartId");
   });
+
+  test("Double clicking only opens composer when actually targetting grid overlay", async () => {
+    // craeting a child  node
```
Comment (hokolomopo) [inline tests/components/grid.test.ts:1476]:
```suggestion
    // creating a child  node
```

File: tests/components/grid.test.ts [hunk 2]
```diff
@@ -1470,4 +1471,24 @@ describe("Copy paste keyboard shortcut", () => {
...
+    mockChart();
+    createChart(model, {}, "chartId");
+    await nextTick();
+    await simulateClick(".o-figure", 0, 0);
+    await nextTick();
+    expect(document.activeElement).toBe(fixture.querySelector(".o-figure"));
+    // click on child
```
Comment (hokolomopo) [inline tests/components/grid.test.ts:1483]:
```suggestion
    // double click on child
```

File: tests/components/grid.test.ts [hunk 3]
```diff
@@ -1470,4 +1471,24 @@ describe("Copy paste keyboard shortcut", () => {
...
+    triggerMouseEvent(".o-figure", "dblclick");
+    await nextTick();
+    expect(model.getters.getEditionMode()).toBe("inactive");
+    expect(document.activeElement).toBe(fixture.querySelector(".o-figure"));
+
+    // click on grid overlay
+    triggerMouseEvent(".o-grid-overlay", "dblclick");
+    await nextTick();
+    expect(document.activeElement).toBe(fixture.querySelector(".o-grid div.o-composer"));
```
Comment (hokolomopo) [inline tests/components/grid.test.ts:1492]: also test editionMode to be consistent with the test just above

**Review by hokolomopo — COMMENTED** [review]
Some typos/nitpick. 

The odoo spec is inclomplete, it says you only fix the paste chart issue, while you do fix other things.

**Review by LucasLefevre — APPROVED** [review]
LGTM
However, the task id in the last two commits doesn't seem to match (the task is about link display)

**Comment by rrahir** [comment]
Actually the second one was also wrong ...

**Review by LucasLefevre — APPROVED** [review]
robodoo r+ rebase-ff

**Pull request closed without merging**